### PR TITLE
PCUI components and observer

### LINKED ISSUE #45: Start viewer with side panel collapsed below a certain width
At small tab widths, the side panel can dominate the tab. If panel width is greater than maybe half the width of the tab, it should start out collapsed.

### LINKED ISSUE #47: Allow the sider bar to be size adjustable
At the moment, some of the morph target names and values are not readable at the default width:
image

### PR DESCRIPTION
**Pull request by ellthompson**
- Used react components from PCUI to build the controls panel
- Used an observer object from PCUI to manage the apps data layer
- Animation controls no longer editable until an animation is present
- Animation play / stop buttons combined
- Centred the initial cta inside the canvas
- Included a js bundle analyzer to monitor the build size
- Control panel can now be resized (Fixes #47)
- Control panel starts closed on small screens (Fixes #45)

image

I confirm I have signed the [Contributor License Agreement](https://docs.google.com/a/playcanvas.com/forms/d/1Ih69zQfJG-QDLIEpHr6CsaAs6fPORNOVnMv5nuo0cjk/viewform).

### CODE REVIEW

File: src/controls.tsx [hunk 1]
```diff
@@ -0,0 +1,140 @@
+// @ts-ignore: library file import
+import { Container, BooleanInput, Label, SliderInput, SelectInput, Panel, Button } from 'lib/pcui/component.js';
+// @ts-ignore: library file import
+import { BindingTwoWay, useObserverState } from 'lib/pcui/binding.js';
+import React, { useEffect, useState, useContext } from 'react';
+import { ProgressPlugin } from 'webpack';
+import { Morph, Option, Observer } from './types';
+
+const ObserverContext = React.createContext(null);
```
Comment (slimbuck) [inline src/controls.tsx:9]: this looks like an instance of something, in which case shouldn't it be named `observerContext` ? Line below too.

**Review by slimbuck — APPROVED** [review]
Looks great!

File: src/controls.tsx (hunk 1 above)
Comment (ellthompson) [reply src/controls.tsx:9]: In '.jsx / .tsx' files react components are generally pascal case to differentiate them from standard DOM elements when used in markup. If you then create an instance of it that component, that'd usually be camel case.

https://airbnb.io/javascript/react/#naming

Comment (slimbuck) [reply src/controls.tsx:9]: gotcha, thanks

**Comment by slimbuck** [comment]
Running the viewer locally I've noticed:
- edit box mouse drag edit doesn't seem to work
- i get console errors loading the model in https://github.com/playcanvas/playcanvas-viewer/issues/55
- this no longer works:
  http://localhost:5000/?load=https://raw.githubusercontent.com/KhronosGroup/glTF-Sample-Models/master/2.0/DamagedHelmet/glTF-Binary/DamagedHelmet.glb

**Comment by ellthompson** [comment]
> Running the viewer locally I've noticed:
> 
> * edit box mouse drag edit doesn't seem to work
> * i get console errors loading the model in #55
> * this no longer works:
>   http://localhost:5000/?load=https://raw.githubusercontent.com/KhronosGroup/glTF-Sample-Models/master/2.0/DamagedHelmet/glTF-Binary/DamagedHelmet.glb

Cheers. The edit box slider should be disabled for slider inputs so i'll switch those off. Also fixed the issue with the load url. I can't seem to reproduce the issue with the SuzanneSequence.glb though (running the latest engine version). Could you paste the error?

**Comment by slimbuck** [comment]
Can confirm edit box slider is fixed and load param works and can't repro Suzanne issue.

**Pull request merged**